pkg/storage/stores/indexshipper/compactor/deletion/delete_requests_manager.go

```diff
@@ -279,7 +290,7 @@ func (d *DeleteRequestsManager) Expired(ref retention.ChunkEntry, _ model.Time)
 		}
 	}
 
-	if len(d.chunkIntervalsToRetain) == 1 && d.chunkIntervalsToRetain[0].Interval.Start == ref.From && d.chunkIntervalsToRetain[0].Interval.End == ref.Through {
```

Comment:
This check here was causing us to skip whole chunks covered by delete requests with line filters